### PR TITLE
Use solver and adjust typing

### PR DESCRIPTION
**Pull request by pacospace**
Signed-off-by: Francesco Murdaca <fmurdaca@redhat.com>

### CODE REVIEW

File: thoth/messaging/unresolved_package.py [hunk 1]
```diff
@@ -39,9 +41,9 @@ class MessageContents(faust.Record, serializer="json"):
         """Class used to represent contents of a message Kafka topic."""
 
         package_name: str
-        package_version: str
-        index_url: str
-        runtime_environment: str
```
Comment (goern) [inline thoth/messaging/unresolved_package.py:44]: why is this removed?

Comment (pacospace) [reply thoth/messaging/unresolved_package.py:44]: better to keep it general? I extracted the information needed for solver from adviser report part related to runtime environment and built the solver (string) that need to be scheduled with priority, if no runtime environment found, all solvers will be scheduled for that package.

```
{'cuda_version': None, 'hardware': {'cpu_family': None, 'cpu_model': None}, 'name': 'rhel:8', 'operating_system': {'name': 'rhel', 'version': '8.0'}, 'python_version': '3.6'}
```

`solver=solver-rhel-8.0-py36`

Comment (pacospace) [reply thoth/messaging/unresolved_package.py:44]: WDYT @goern ?
